workers/unity/Assets/Playground/Scripts/Worker/WorkerConnectorBase.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Improbable.Gdk.Core;
+using Improbable.Worker.Core;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Playground
+{
+    public abstract class WorkerConnectorBase : MonoBehaviour, IDisposable
+    {
+        private delegate Task<Worker> ConnectionDelegate();
+
+        public GameObject LevelPrefab;
+        public int MaxConnectionAttempts = 3;
+        public bool UseExternalIp = false;
+
+        public Worker Worker;
+
+        protected WorkerConnectorBase[] RequiredWorkerConnection;
+
+        private GameObject levelInstance;
+
+        private readonly TaskCompletionSource<bool> connectionAttemptFinishedTask = new TaskCompletionSource<bool>();
+
+        // Important run in this step as otherwise it can interfere with the the domain unloading logic
+        private void OnApplicationQuit()
+        {
+            Dispose();
+        }
+
+        private void OnDestroy()
+        {
+            Dispose();
+        }
+
+        public async Task Connect(string workerType, ILogDispatcher logger)
+        {
+            // Check that other workers have finished trying to connect before this one starts
+            // This prevents races on the workers starting and races on when we start icking systems
+            if (RequiredWorkerConnection != null)
+            {
+                Task[] requiredWorkers = RequiredWorkerConnection
+                    .Select(connection => connection.connectionAttemptFinishedTask.Task).ToArray();
+                try
+                {
+                    await Task.WhenAll(requiredWorkers);
+                }
+                catch (Exception e)
+                {
+                    logger.HandleLog(LogType.Error,
+                        new LogEvent("Cancelling connection attempt")
+                            .WithField("WorkerType", workerType)
+                            .WithField("Reason", "Required worker failed to connect"));
+                    connectionAttemptFinishedTask.SetException(e);
+                    Dispose();
+                    return;
+                }
+            }
+
+            try
+            {
+                var origin = transform.position;
+                if (ShouldUseLocator())
+                {
+                    var config = GetLocatorConfig(workerType);
```

Comment:
if we are going to move this to Core, we need to make getting the configs overridable. with mobile for example we can't parse command line arguments, but must pass the values in differently (e.g. through a splash screen)

Reply:
This isn't in core.